mesonbuild/compilers/detect.py

```diff
@@ -648,6 +648,10 @@ def detect_cuda_compiler(env: 'Environment', for_machine: MachineChoice) -> Comp
         cpp_compiler = detect_cpp_compiler(env, for_machine)
         cls = CudaCompiler
         env.coredata.add_lang_args(cls.language, cls, for_machine, env)
+        for k in env.options:
+            if 'cuda' in k.name:
```

Comment:
```suggestion
            if 'cuda' in k.name and k.machine is for_machine:
```

We've had talk about using `clang` as a cuda compiler, let's only touch up the cuda arguments for the machine we're working with.

Reply:
And actually, this isn't safe anymore. We would have been able to do `k.language == 'cuda'` and gotten a safe answer, but that isn't true anymore because someone could have an option like `my_cuda_option`. So the only guaranteed way to check this is to do:
```python
if k.name.startswith('cuda_') and k.machine is for_machine:
```

Reply:
Changed accordingly.